plugins/outputs/microsoft_fabric/event_stream.go

```diff
@@ -0,0 +1,189 @@
+//go:generate ../../../tools/readme_config_includer/generator
+package microsoft_fabric
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/messaging/azeventhubs"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/config"
+	"github.com/influxdata/telegraf/internal"
+	"github.com/influxdata/telegraf/plugins/serializers/json"
+)
+
+type eventstream struct {
+	connectionString string
+	timeout          config.Duration
+	log              telegraf.Logger
+
+	partitionKey   string
+	maxMessageSize config.Size
+
+	client     *azeventhubs.ProducerClient
+	options    azeventhubs.EventDataBatchOptions
+	serializer telegraf.Serializer
+}
+
+func (e *eventstream) init() error {
+	// Parse the connection string by splitting it into key-value pairs
+	// and extract the extra keys used for plugin configuration
+	pairs := strings.Split(e.connectionString, ";")
+	for _, pair := range pairs {
+		// Split each pair into key and value
+		k, v, found := strings.Cut(pair, "=")
+		if !found {
+			return fmt.Errorf("invalid connection string format: %q", pair)
+		}
+
+		// Only lowercase the keys as the values might be case sensitive
+		k = strings.ToLower(strings.TrimSpace(k))
+		v = strings.TrimSpace(v)
+
+		key := strings.ReplaceAll(k, " ", "")
+		switch key {
+		case "partitionkey":
+			e.partitionKey = v
+		case "maxmessagesize":
+			msgsize, err := strconv.ParseUint(v, 10, 64)
+			if err != nil {
+				return fmt.Errorf("invalid max message size: %w", err)
+			}
+			if msgsize > 0 {
+				e.options.MaxBytes = msgsize
+			}
+		}
+	}
+
+	// Setup the JSON serializer
+	serializer := &json.Serializer{
+		TimestampUnits:  config.Duration(time.Nanosecond),
+		TimestampFormat: time.RFC3339Nano,
+	}
+	if err := serializer.Init(); err != nil {
+		return fmt.Errorf("setting up JSON serializer failed: %w", err)
+	}
+	e.serializer = serializer
+
+	return nil
+}
+
+func (e *eventstream) Connect() error {
+	cfg := &azeventhubs.ProducerClientOptions{
+		ApplicationID: internal.FormatFullVersion(),
+		RetryOptions:  azeventhubs.RetryOptions{MaxRetries: -1},
+	}
+
+	client, err := azeventhubs.NewProducerClientFromConnectionString(e.connectionString, "", cfg)
+	if err != nil {
+		return fmt.Errorf("failed to create client: %w", err)
+	}
+	e.client = client
+
+	return nil
+}
+
+func (e *eventstream) Close() error {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(e.timeout))
+	defer cancel()
+
+	return e.client.Close(ctx)
+}
```

Comment:
No nil check - will panic if e.client is nil
```suggestion
func (e *eventstream) Close() error {
	if e.client == nil {
		return nil
	}
	
	ctx, cancel := context.WithTimeout(context.Background(), time.Duration(e.timeout))
	defer cancel()

	return e.client.Close(ctx)
}
```